pkg/controller/deployment/deployment_controller.go

```diff
@@ -356,11 +356,21 @@ func (dc *DeploymentController) deletePod(obj interface{}) {
 	glog.V(4).Infof("Pod %s deleted.", pod.Name)
 	if d := dc.getDeploymentForPod(pod); d != nil && d.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
 		// Sync if this Deployment now has no more Pods.
-		rsList, err := dc.getReplicaSetsForDeployment(d)
+		selector, err := metav1.LabelSelectorAsSelector(d.Spec.Selector)
 		if err != nil {
 			return
 		}
-		podMap, err := dc.getPodMapForDeployment(d, rsList)
+		rsList, err := dc.rsLister.ReplicaSets(d.Namespace).List(selector)
```

Comment:
Use [ListReplicaSets](https://github.com/kubernetes/kubernetes/blob/8df56da448fafdacbacd5139315afc4b5ef752a3/pkg/controller/deployment/util/deployment_util.go#L557).

Reply:
The code submitted here, why does it need to obtain the replicaSet from apiserver, why not obtain it through rsLister from indexer